lib/meandro/rules/unused_callbacks.ex

```diff
@@ -26,16 +32,49 @@ defmodule Meandro.Rule.UnusedCallback do
     false
   end
 
-  defp analyze_file(file, _ast) do
-    [set_result(file, nil, nil, nil)]
+  defp analyze_file(file, ast) do
+    {_, callbacks} = Macro.prewalk(ast, [], &collect_callbacks/2)
+
+    for {name, arity, line} <- callbacks, is_unused(name, arity, ast) do
```

Comment:
this is a comment just to make you consider counting the usage of each callback already when you collect them, so we don't have to walk the AST multiple times to get the count afterwards

Reply:
I think the code is clearer this way. What you suggest is totally possible, but I think it will make the code uglier and harder to understand.